src/Symfony/Bundle/FrameworkBundle/Command/SecretsGenerateKeyCommand.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Exception\EncryptionKeyNotFoundException;
+use Symfony\Bundle\FrameworkBundle\Secret\EncoderInterface;
+use Symfony\Bundle\FrameworkBundle\Secret\MutableSecretStorageInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class SecretsGenerateKeyCommand extends Command
+{
+    protected static $defaultName = 'secrets:generate-key';
+    private $secretsStorage;
+    private $encoder;
+
+    public function __construct(EncoderInterface $encoder, MutableSecretStorageInterface $secretsStorage)
```

Comment:
If I'm right, we don't use prefixes like `Mutable` and `Immutable` in Symfony code.

Reply:
I would drop the prefix as well

Reply:
naming things are hard... :(

Here I wanted to distingquihed "readonly storage" from "writable storage" do you have any idea of naming?

note: we have a `Symfony\Component\Security\Acl\Model\MutableAclInterface`

Reply:
```php
SecretStorageInterface {
    get(): string;
    set(): void;
    list(): iterable;
}
```
would be enough IMHO

Reply:
👎 don't just name it get, set etc without "secret". It should be obvious from the method name that this is confidential data, not a generic data store.

Reply:
given:
- At first, users only need to fetch secrets from the storage. That's the purpose of the `SecretStorageInterface`.
- Symfony provides a default storage (the `FileSecretStorage`) and a convenient way to add new secrets in it (the `SecretsAddCommand`). Maybe I should inject the `FileSecretStorage` and remove that interface?

We have 2 interface for reasons:

1) From our point of view, it didn't make sens to use symfony's command to manage 3rd party tools like `vault`. Users could use their favorite tool to add/delete secrets from them.

2) I think that a futur implementation of `SecretStorageInterface` could be a `ChainSecretStorage` that iterate over a list of `SecretStorageInterface`. Implementing `setSecret` in a chain wouldn't make sens.

Reply:
Added the ChainSecretStorage which will improve the DX by allowing developper/3rd party to create new SecretStorages